lms/static/scripts/frontend_apps/utils/test/youtube-test.js

```diff
@@ -0,0 +1,74 @@
+import { validateYouTubeVideoURL, videoIdFromYouTubeURL } from '../youtube';
+
+describe('youtube', () => {
+  describe('videoIdFromYouTubeURL', () => {
+    [
+      { url: 'https://youtu.be/cKxqzvzlnKU', expectedId: 'cKxqzvzlnKU' },
+      { url: 'https://www.youtube.com/watch?v=123', expectedId: '123' },
+      {
+        url: 'https://www.youtube.com/watch?channel=hypothesis&v=foo',
+        expectedId: 'foo',
+      },
+      {
+        url: 'https://www.youtube.com/embed/embeddedId',
+        expectedId: 'embeddedId',
+      },
+      {
+        url: 'https://www.youtube.com/shorts/shortId',
+        expectedId: 'shortId',
+      },
+      {
+        url: 'https://www.youtube.com/live/liveId?feature=share',
+        expectedId: 'liveId',
+      },
+    ].forEach(({ url, expectedId }) => {
+      it('resolves ID from valid YouTube video', () => {
+        assert.equal(videoIdFromYouTubeURL(url), expectedId);
+      });
+    });
+  });
+
+  describe('validateYouTubeVideoUrl', () => {
+    [
+      {
+        url: 'foo',
+        expectedError:
+          'Please enter a YouTube URL, e.g. "https://www.youtube.com/watch?v=cKxqzvzlnKU"',
```

Comment:
I used this approach because it was being used in `URLPicker` (it was my first reference), but I don't have a preference one way or another. I wIll change them.